R/readWQPdata.R

```diff
@@ -147,35 +147,40 @@ readWQPdata <- function(..., zip=FALSE, querySummary=FALSE){
 
   values <- sapply(values, function(x) URLencode(x, reserved = TRUE))
 
-  urlCall <- paste(paste(names(values),values,sep="="),collapse="&")
+  # add special parameters
+  values[["sorted"]] <- "no"
+  values[["mimeType"]] <- "tsv"
+  if (zip) values[["zip"]] <- "yes"
   
-  baseURL <- drURL("wqpData")
-  urlCall <- paste0(baseURL,
-                   urlCall,
-                   "&sorted=no&mimeType=tsv")
-  
-  if(zip) urlCall <- paste0(urlCall,"&zip=yes")
+  urlCall <- drURL("wqpData", arg.list=values)
```

Comment:
I switched to this as I needed to build two different urls and didn't want to repeat the above.